include/xtensor/xpad.hpp

```diff
@@ -251,7 +251,7 @@ namespace xt
 
         XTENSOR_ASSERT(e.shape().size() == reps.size());
 
-        auto new_shape = e.shape();
+        typename return_type::shape_type new_shape;
```

Comment:
If shape_type is a `std::vector`, line 258 will crash because the `vector` has not been resized.